art/attacks/evasion/dpatch.py

```diff
@@ -108,8 +108,29 @@ def generate(
         :param x: Sample images.
         :param y: Target labels for object detector.
         :param target_label: The target label of the DPatch attack.
+        :param mask: An boolean array of shape equal to the shape of a single samples (1, H, W) or the shape of `x`
+                     (N, H, W) without their channel dimensions. Any features for which the mask is True can be the
+                     center location of the patch during sampling.
+        :type mask: `np.ndarray`
         :return: Adversarial patch.
         """
+        mask = kwargs.get("mask")
```

Comment:
Line 117-133: This code is also similar to that in adversarial_patch_numpy/tensorflow. Can we put them in a utility function?

Reply:
I agree, we should unify the checks across all attacks.